### PR TITLE
James 2286 : covers current implementations and drop unsatisfying ones

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by chibenwa** [comment]
I would propose **a vote** as this is some serious decision for the project.

Maybe we can delay the "dropping" part, but merge the rest?

**Comment by chibenwa** [comment]
test this please

OutOfMemory Java heap space

File: server/data/data-file/src/test/java/org/apache/james/mailrepository/FileMailRepositoryTest.java [hunk 1]
```diff
@@ -25,32 +25,85 @@
...
+        }
+
+        @Override
+        public MailRepository retrieveRepository() {
+            return mailRepository;
+        }
+    }
+
+    @Nested
+    @DisplayName("Default configuration")
```
Comment (chibenwa) [inline server/data/data-file/src/test/java/org/apache/james/mailrepository/FileMailRepositoryTest.java]: My choice went to **Nested** as over generic extensions Akka `@TestTemplate` and `@ParameterizedTest` are to be specified on each test.

As I want it to be used on the entire class, this is the cleanest solution I found.

**Comment by chibenwa** [comment]
test this please

```
OutOfMemory Java heap space
```

**Comment by mbaechler** [comment]
> I would propose a vote as this is some serious decision for the project.

👍

**Comment by mbaechler** [comment]
> Maybe we can delay the "dropping" part, but merge the rest?

👍  too

File: server/data/data-file/src/test/java/org/apache/james/mailrepository/FileMailRepositoryTest.java (hunk 1 above)
Comment (mbaechler) [reply server/data/data-file/src/test/java/org/apache/james/mailrepository/FileMailRepositoryTest.java]: not bad

**Comment by mbaechler** [comment]
I removed the removal and will write a mail to the mailing list about MBox and JCR removal.

**Pull request merged**